saltwater-repl/repl.rs

```diff
@@ -0,0 +1,231 @@
+use crate::{
+    commands::default_commands, commands::generate_help_message, commands::Command,
+    helper::ReplHelper,
+};
+use dirs_next::data_dir;
+use rustyline::{error::ReadlineError, Cmd, CompletionType, Config, EditMode, Editor, KeyPress};
+use saltwater_codegen::{compile_decl, initialize_jit_module, JIT};
+use saltwater_parser::{
+    data, hir, hir::Declaration, hir::Expr, types, CompileError, Locatable, Parser,
+    PreProcessorBuilder, PureAnalyzer, Type,
+};
+use std::path::PathBuf;
+
+/// The prefix for commands inside the repl.
+pub(crate) const PREFIX: char = ':';
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const PROMPT: &str = ">> ";
+
+/// Takes a function pointer and transmute it into a function
+/// that returns the given type. Then executes it and returns the result.
+///
+/// # Safety
+///
+/// `$fun` has to be a valid pointer to a function that returns the `$ty`.
+macro_rules! execute {
+    ($fun:ident, $ty:tt) => {
+        unsafe {
+            let execute: unsafe extern "C" fn() -> $ty = std::mem::transmute($fun);
```

Comment:
Let's also specify the source type.
```suggestion
            let execute = std::mem::transmute<*const u8, unsafe extern "C" fn() -> $ty> ($fun);
```